src/AutoRest.CSharp/Mgmt/Decorator/ReferenceTypePropertyChooser.cs

```diff
@@ -77,31 +77,25 @@ public static bool TryGetCachedExactMatch(Schema schema, out CSharpType? result)
             return null;
         }
 
-        private static ObjectTypeProperty? FindSimpleReplacements(ObjectTypeProperty originalType, Type frameworkType, BuildContext context)
+        private static ObjectTypeProperty? FindSimpleReplacements(ObjectTypeProperty originalType, Type frameworkType)
         {
             //TODO for core generation this list is small enough we can simply define each of them here.
             //eventually we might want to come up with a more robust way of doing this
 
             bool isString = frameworkType == typeof(string);
 
             if (originalType.Declaration.Name == "Location" && (isString || frameworkType.Name == _locationType.Name))
-                return GetObjectTypeProperty(originalType, _locationType, context);
+                return GetObjectTypeProperty(originalType, _locationType);
 
             if (originalType.Declaration.Name == "ResourceType" && (isString || frameworkType.Name == _resourceTypeType.Name))
-                return GetObjectTypeProperty(originalType, _resourceTypeType, context);
+                return GetObjectTypeProperty(originalType, _resourceTypeType);
 
             if (originalType.Declaration.Name == "Id" && (isString || frameworkType.Name == _resourceIdentifierType.Name))
-                return GetObjectTypeProperty(originalType, _resourceIdentifierType, context);
+                return GetObjectTypeProperty(originalType, _resourceIdentifierType);
```

Comment:
These three lines are wrapping `ResourceType` `AzureLocation` and `ResourceIdentifier` into a `SystemObjectType` which is unnecessary and will lead to wrong serialization and deserialization method when generating resourcemanager.
Therefore here I simplified this to use replace the type using the corresponding Framework type